modules/@apostrophecms/piece-type/index.js

```diff
@@ -1064,6 +1078,18 @@ module.exports = {
           }
           return true;
         });
+      },
+      getManagerApiProjection(req) {
+        if (!self.options.managerApiProjection) {
+          return null;
+        }
+
+        const projection = { ...self.options.managerApiProjection };
+        self.columns.forEach(({ name }) => {
```

Comment:
Hmm. Not enough yet. Consider:

```
      'draft:submitted.at': {
        label: 'apostrophe:submitted',
        component: 'AposCellLastEdited'
      },
      'draft:submitted.by': {
        label: 'apostrophe:user',
        component: 'AposCellBasic'
      }
```

`draft:submitted.by` pulls the `by` subproperty of the `submitted` property.

If the name contains a `draft:` or `published:` part, remove that part. It's not related to the projection, so not your problem.

Keep the rest as the projection path and it should work.

You can confirm success by looking at the "submitted draft" manager which uses this feature to show who submitted a draft and when.

Technically a column can have a custom component and display any property of any item. It isn't possible for us to find all of the properties they might pull and include them in the projection. However this is probably rare so far, and I think users who are sophisticated enough to be writing those will be OK with adding to the projection. If we do get complaints we may have to add an option to enable the manager projection feature.